tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -11,10 +11,10 @@
 import re
 
 # 'az' space then repeating runs of quoted tokens or non quoted characters
-_az_pattern = 'az\s*' + '(([^\"\'])*|' + '((\"[^\"]*\"\s*)|(\'[^\']*\'\s*))' + ')'
+_az_pattern = r'az\s*' + '(([^\"\'])*|' + '((\"[^\"]*\"\\s*)|(\'[^\']*\'\\s*))' + ')'
 # match the two types of command substitutions
-_CMD_SUB_1 = re.compile("\$\(\s*" + "(" + _az_pattern + ")" + "\)")
-_CMD_SUB_2 = re.compile("`\s*" + "(" + _az_pattern + ")" + "`")
+_CMD_SUB_1 = re.compile(r"\$\(\s*" + "(" + _az_pattern + ")" + r"\)")
+_CMD_SUB_2 = re.compile(r"`\s*" + "(" + _az_pattern + ")" + "`")
```

Comment:
I have two questions about this PR:
1. Some of the regular expressions are not using the raw string (r) mode.
2. To ensure the behavior remains consistent before and after the change, could you add some additional tests to verify these modifications?